### DISCUSSION TITLE
How To: Configuring Ender 5 Plus Stock Screen with BTK SKR Mini E3 V2.0 Plus Print Progress Display

### DISCUSSION THREAD

**Discussion by ghost** [Desuuuu/klipper#117] [General]
After countless hours of research, you will need to reconfigure the wire that connects to the screen for the BTK SKR Mini E3 V2.0 to work and set the Screen Serial port to USART1 based on the diagram from here: 

https://www.deviousweb.com/2020/05/08/ender-5-plus-skr1-3-and-factory-screen/

https://imgur.com/a/8JN3NYT

https://github.com/bigtreetech/BIGTREETECH-SKR-mini-E3/blob/master/hardware/BTT%20SKR%20MINI%20E3%20V2.0/Hardware/BTT%20SKR%20MINI%20E3%20V2.0-PIN.pdf

then if you want Print Progress display to work on klipper, 
you would want to configure your macros to update constantly as such where you'd want to enable the loop at the start of the print and stop at the end of the print
```
[gcode_macro START_PRINT]
gcode:
   DGUS_PRINT_START
   UPDATE_DELAYED_GCODE ID=DISPLAY_UPDATE DURATION=2
   ...

[gcode_macro END_PRINT]
gcode:
   UPDATE_DELAYED_GCODE ID=DISPLAY_UPDATE DURATION=0
   DGUS_PRINT_END
   ...

[delayed_gcode DISPLAY_UPDATE]
gcode:
   M73 P{"%2.0f" % (printer.virtual_sdcard.progress * 100)}
   UPDATE_DELAYED_GCODE ID=DISPLAY_UPDATE DURATION=2
```

this is the most roundabout way I can provide to everyone who is also doing the same for their skr mini e3 v2.0
I tried getting it to work on the 3.0 but I've run into some compiler issues that I just do not have the time to fix debugging, so use the 2.0 for compatibility sake.

**Comment by napter** [comment]
Hey - how do you set the screen serial port to "USART1"?

**Reply by ghost** [reply]
It's configured under klipper menu config when compiling the firmware
